src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerAssemblyLoader.cs

```diff
@@ -93,8 +98,31 @@ internal abstract partial class AnalyzerAssemblyLoader : IAnalyzerAssemblyLoader
         /// </summary>
         private partial bool IsMatch(AssemblyName requestedName, AssemblyName candidateName);
 
+        private void CheckIfDisposed()
```

Comment:
Can we use DIM in the compiler layer? We still need to run on Framework, which doesn't support it, right?